app/controllers/users_controller.rb

```diff
@@ -1,54 +1,21 @@
 class UsersController < ApplicationController
   before_action :set_user, only: [:show, :edit, :update, :destroy]
-  before_filter :check_if_verified, only: [:finish_signup]
-  # GET /users/:id.:format
-  def show
-    # authorize! :read, @user
-  end
+  before_filter :check_current_user, only: [:update]
 
-  # GET /users/:id/edit
-  def edit
-    # authorize! :update, @user
+  def me 
+    render json: current_user.to_json
   end
 
-  # PATCH/PUT /users/:id.:format
   def update
-    # authorize! :update, @user
-    respond_to do |format|
-      if @user.update(user_params)
-        sign_in(@user == current_user ? @user : current_user, :bypass => true)
-        format.html { redirect_to @user, notice: 'Your profile was successfully updated.' }
-        format.json { head :no_content }
-      else
-        format.html { render action: 'edit' }
-        format.json { render json: @user.errors, status: :unprocessable_entity }
-      end
-    end
-  end
+    data = user_params
+    missing_fields = fields.select { |f| data[f].nil? }
```

Comment:
Yes, it is a validation, given that the following fields where requested of the user, he must fill every field, I don't see another location to do this validation except for the client, but in my opinion is better to validate on the server than on the client.